music21/noteworthy/translate.py

```diff
@@ -453,20 +486,39 @@ def translateNote(self, attributes):
         durationInfo = attributes['Dur']
         pitchInfo = attributes['Pos']
 
-        n = note.Note()   # note!
+        p = self.getPitchFromPositionInfo(pitchInfo)
+
+        if self.currentClef == 'PERCUSSION':
+            n = note.Unpitched()
+            n.fromDisplayPitch(p)
+        else:
+            n = note.Note()   # note!
+            n.pitch = p
 
         # durationInfo
         self.setDurationForObject(n, durationInfo)
 
         # pitchInfo
         self.setTieFromPitchInfo(n, pitchInfo)
-        n.pitch = self.getPitchFromPositionInfo(pitchInfo)
 
         # if Lyrics
         if self.lyrics and self.lyricPosition < len(self.lyrics):
             n.addLyric(self.lyrics[self.lyricPosition])
 
-        self.currentMeasure.append(n)
+        self.getCurrentVoice().append(n)
+
+    # current voice has shorter duration. If no voices, return current measure
+    def getCurrentVoice(self):
```

Comment:
needs docs, and return typing.

Reply:
Especially since it doesn't necessarily return a stream.Voice object as the name implies.

Reply:
as this is now called for every note, it needs to be optimized as much as possible.  I'm seeing most optimized version as:

```
if self.currentMeasure.isFlat:
     # O(1) in most cases.
     return self.currentMeasure

# O(n) only if voices already exist.
return min(
     self.currentMeasure.getElementsByClass(stream.Voice), 
     key=lambda v: v.quarterLength
)
```